crates/re_space_view_spatial/src/ui_3d.rs

```diff
@@ -266,24 +273,25 @@ pub fn view_3d(
     ctx: &mut ViewerContext<'_>,
     ui: &mut egui::Ui,
     state: &mut SpatialSpaceViewState,
-    space: &EntityPath,
-    space_view_id: SpaceViewId,
-    scene: &mut SceneSpatial,
-) {
+    view_ctx: &ViewContextCollection,
+    parts: &ViewPartCollection,
+    query: &ViewQuery<'_>,
+    mut draw_data: Vec<re_renderer::QueueableDrawData>,
+) -> Result<(), SpaceViewSystemExecutionError> {
     re_tracing::profile_function!();
 
-    let highlights = &scene.highlights;
-    let space_cameras = &scene.parts.cameras.space_cameras;
+    let highlights = query.highlights;
+    let space_cameras = &parts.get::<CamerasPart>()?.space_cameras;
     let view_coordinates = ctx
```

Comment:
Nothing to do here in this PR, but just observing doing a query here actually feels kind of weird. Should this have been done as part of a ViewContext maybe?

Reply:
you mean the fact that we get `ViewQuery` in in the first place? Yeah would be nice to restrict the actual store access to the systems. `ViewQuery` lists out quite a bit of data now that is useful without actually doing queries which is also a lil bit questionable